src/main/java/com/github/krzysiekjodlowski/ox/Board.java

```diff
@@ -0,0 +1,110 @@
+package com.github.krzysiekjodlowski.ox;
+
+import java.util.HashMap;
+import java.util.Map;
+
+/**
+ * Represents square game board.
+ * @author KrzysiekJodlowski
+ */
+class Board {
+    /**
+     * Size of the board game.
+     */
+    private final int boardCapacity;
+    /**
+     * Minimal boardSize.
+     */
+    private final int minBoardSize = 3;
+    /**
+     * Maximal boardSize (temporary value for now).
+     */
+    private final int maxBoardSize = 1000;
+    /**
+     * Core field in this class.
+     */
+    private final Map<FieldNumber, Symbol> moves = new HashMap<>();
+    /**
+     * Size of one side of the board.
+     */
+    private final int boardSize;
+
+    /**
+     *  Constructor initialization.
+     * @param boardSize defines one board game side length
+     * @throws BoardSizeOutOfBoundsException exception
+     * which should be handled by client
+     */
+    Board(final int boardSize) throws BoardSizeOutOfBoundsException {
+        this.checkBoardSize(boardSize);
+        this.boardSize = boardSize;
+        this.boardCapacity = boardSize * boardSize;
+    }
+
+    /**
+     * Simple check if boardSize parameter meets the conditions.
+     * @param boardSize passed by client when creating a Board
+     * @throws BoardSizeOutOfBoundsException exception
+     * which should be handled by client
+     */
+    private void checkBoardSize(final int boardSize)
+            throws BoardSizeOutOfBoundsException {
+        if (boardSize < this.minBoardSize) {
+            throw new BoardSizeOutOfBoundsException(
+                    String.format("boardSize should be greater than %d!",
+                    this.minBoardSize));
```

Comment:
If you change minBoardSize to static, you will have to remove all 'this' references.

Reply:
Now there is no max / min values for the size - in the future it will be placed in Settings.